metronome/hotstuff/service/test/src/io/iohk/metronome/hotstuff/service/ConsensusServiceSpec.scala

```diff
@@ -0,0 +1,121 @@
+package io.iohk.metronome.hotstuff.service
+
+import io.iohk.metronome.hotstuff.consensus.basic.Agreement
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.compatible.Assertion
+import org.scalatest.flatspec.AsyncFlatSpec
+import scala.concurrent.duration._
+import org.scalatest.matchers.should.Matchers
+import io.iohk.metronome.hotstuff.consensus.basic.{
+  ProtocolError,
+  Event,
+  Message,
+  Phase,
+  QuorumCertificate
+}
+import io.iohk.metronome.hotstuff.consensus.ViewNumber
+import io.iohk.metronome.crypto.GroupSignature
+
+class ConsensusServiceSpec extends AsyncFlatSpec with Matchers {
```

Comment:
Are we going to add more tests here once we fill in the missing bits in `ConsensusService`? Regardless, since this tests only the `MessageStash` maybe it's better have be just about that and rename to `MessageStashSpec`?

Reply:
I haven't added more tests yet. Although I feel a little bad about that, it looked like it's more trouble than its worth, and wanted to just do an integration test that somehow covers the whole system behaviour, or to see how it can be refactored into smaller components like this `MessageStash` that are easy to test.

I thought since it's a nested class it can be inside the `ConsensusServiceSpec` and this `behave like` makes it easy to see what belongs to it, the code can be collapsed as one region as well. 

But if you think it's better to have it in a separate class, that's fine with me.